Adafruit_DAP_SAM.cpp

```diff
@@ -112,14 +116,18 @@ device_t Adafruit_DAP_SAM::devices[] = {
     {0},
 };
 
+
 //-----------------------------------------------------------------------------
 bool Adafruit_DAP_SAM::select(uint32_t *found_id) {
   uint32_t DAP_DSU_did;
 
-  // Stop the core
-  dap_write_word(DHCSR, 0xa05f0003);
-  dap_write_word(DEMCR, 0x00000001);
-  dap_write_word(AIRCR, 0x05fa0004);
+  // Stopping the core fails on my Arduino MKR board with SAM D21G18A (Rev D), when not doing an Extended reset first.
```

Comment:
we should really implement `reset_with_extension()` as member function and invoke it as part of the target setletct as it does with the edbg commit https://github.com/ataradov/edbg/commit/e07b65baf2bd9e0c1db1445738e1fb72f63607e8#diff-7f767e69330208cee16f6e5e29ddca182d33ebaa0fe2f6498276f1d70e2bbdc1R173